Sources/MlemMiddleware/API Client/Helpers/PreloadImages.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  PreloadImages.swift
+//
+//
+//  Created by Eric Andrews on 2024-07-07.
+//
+
+import Foundation
+import Nuke
+
+func preloadImages(
```

Comment:
Could this be a method of `Post2Providing`?

Reply:
It could, but it feels a little weird to put it there since it's a bulk operation. There is a bunch of `Post2`-specific logic in here, though--IMO it would make more sense to keep this function freestanding but have it just take in `[ImageRequest]` instead of `[Post2]` and put the `ImageRequest` generation code in `Post2Providing`

Reply:
Ok, that makes sense 👍

I just read this in the [Nuke docs](https://kean-docs.github.io/nuke/documentation/nuke/prefetching/): 

> The prefetcher automatically cancels all of the outstanding tasks when deallocated. All ImagePrefetcher methods are thread-safe and are optimized to be used even from the main thread during scrolling.

Does this mean that the prefetching doesn't actually have time to fetch all of the images when used like this? If so, it might be better to save it as a property of the feed loader.

```swift
// In the feed loader
let requests = posts.flatMap {
    $0.imageRequests(smallAvatarSize: smallAvatarSize, largeAvatarSize: largeAvatarSize)
} 
prefetcher.startPrefetching(with: requests)
```

Reply:
Ooh, that's a good point. With the image requests kicked down to `Post2Providing` it's also trivial enough to inline this that we don't need the standalone helper.